src/net/db.rs

```diff
@@ -1073,6 +1263,149 @@ impl PeerDB {
                     WHERE network_id = ?11 AND addrbytes = ?12 AND port = ?13", args)
             .map_err(db_error::SqliteError)?;
 
+        if let Some(old_peer) = old_peer_opt {
+            let slots = Self::find_peer_slots(
+                tx,
+                neighbor.addr.network_id,
+                &neighbor.addr.addrbytes,
+                neighbor.addr.port,
+            )?;
+            if old_peer.public_key.to_bytes_compressed()
+                != neighbor.public_key.to_bytes_compressed()
+            {
+                // this peer has re-keyed, so it might be a new peer altogether.
+                // require it to re-announce its DBs
+                for slot in slots {
+                    debug!("Peer at slot {} changed; dropping its DBs", slot);
+                    PeerDB::drop_stacker_dbs(tx, slot)?;
+                }
+            }
+        }
+        Ok(())
+    }
+
+    /// Find a peer's slots
+    fn find_peer_slots(
+        conn: &Connection,
+        network_id: u32,
+        addrbytes: &PeerAddress,
+        port: u16,
+    ) -> Result<Vec<u32>, db_error> {
+        let slots = PeerDB::peer_slots(conn, network_id, addrbytes, port)?;
+        let mut ret = vec![];
+        for slot in &slots {
+            let used_slot = PeerDB::has_peer_at(conn, network_id, *slot)?;
+            if used_slot {
+                ret.push(*slot);
+            }
+        }
+        Ok(ret)
+    }
+
+    /// Get stacker DBs for a slot
```

Comment:
Bump on this -- I think the usage of "stacker db" in the code currently is very confusing. It's being used ambiguously to refer both the data being tracked and the contract identifier of the contract governing that data. There are many places where a comment says something like "get stacker dbs for a slot" but it means something more like "get the ContractIDs whose associated databases are being tracked by the peers store in the slots `used_slots`" (because a "slot" is actually itself a reference). The methods and fields in this file need better rustdocs.